batik-bridge/src/main/java/org/apache/batik/bridge/SVGBufferedImage.java

```diff
@@ -73,13 +75,31 @@ public Element getSvgNode() {
         Source source = new DOMSource(svgNode);
         ByteArrayOutputStream out = new ByteArrayOutputStream();
         Result result = new StreamResult(out);
-        TransformerFactory factory = TransformerFactory.newInstance();
+        TransformerFactory factory = getDefaultTransformerFactory();
         Transformer transformer = factory.newTransformer();
         transformer.setOutputProperty("omit-xml-declaration", "yes");
         transformer.transform(source, result);
         return out.toByteArray();
     }
 
+    public static TransformerFactory getDefaultTransformerFactory() {
+        TransformerFactory transformerFactory = TransformerFactory.newInstance();
+        // For saxon or xalan transformer factory it is impossible to set external DTD, doctype declaration
+        // and stylesheet properties
+        if ("net.sf.saxon.TransformerFactoryImpl".equals(transformerFactory.getClass().getCanonicalName())
+            || transformerFactory instanceof org.apache.xalan.processor.TransformerFactoryImpl) {
+            return transformerFactory;
+        }
+        try {
+            transformerFactory.setFeature("http://apache.org/xml/features/disallow-doctype-decl", true);
+        } catch (TransformerConfigurationException e) {
+            System.err.println("could not apply security feature to the transformer factory" + e.getMessage());
```

Comment:
As strange as it seems, batik does not use any logging framework. Protection code for TransformerFactory adapted from DocFamily